targets/TARGET_STM/TARGET_STM32WL/i2c_device.h

```diff
@@ -34,31 +42,14 @@ extern "C" {
 #define I2CAPI_I2C3_CLKSRC RCC_I2C3CLKSOURCE_SYSCLK
 
 /*  Provide the suitable timing depending on requested frequency */
-static inline uint32_t get_i2c_timing(int hz)
-{
-    uint32_t tim = 0;
-
-    // Common settings: I2C clock = 48 MHz, Analog filter = ON, Digital filter coefficient = 0
-    switch (hz) {
-        case 100000:
-            tim = 0x20E03F53; // Standard mode with Rise Time = 640ns and Fall Time = 20ns
-            break;
-        case 400000:
-            tim = 0x20500817; // Fast mode with Rise Time = 250ns and Fall Time = 100ns
-            break;
-        case 1000000:
-            tim = 0x00500A18; // Fast mode Plus with Rise Time = 60ns and Fall Time = 100ns
-            break;
-        default:
-            break;
-    }
-    return tim;
-}
+uint32_t I2C_GetPclk(I2CName i2c);
+uint32_t I2C_ComputeTiming(uint32_t clock_src_freq, uint32_t i2c_freq);
+uint32_t get_i2c_timing(I2CName i2c, int hz);
+void I2C_Compute_PRESC_SCLDEL_SDADEL(uint32_t clock_src_freq, uint32_t I2C_speed);
+uint32_t I2C_Compute_SCLL_SCLH(uint32_t clock_src_freq, uint32_t I2C_speed);
```

Comment:
The function declarations using camelcase or just as the rest of HAL only lower case, can we unify them?

I assume i2c_get_pclk() or if you want to keep this different from Mbed OS HAL, you can add a prefix for instance.

Reply:
@0xc0170 I think its better to change all to `**camelCase**` format.

Reply:
@0xc0170 This will be applicable for all files rite.

Reply:
@0xc0170 I have modified function name to below mentioned. Hope this will make it unique. Update me if this is okay. 

```
uint32_t get_i2c_pclk(I2CName i2c);
uint32_t i2c_compute_timing(uint32_t clock_src_freq, uint32_t i2c_freq);
uint32_t get_i2c_timing(I2CName i2c, int hz);
void i2c_compute_Presc_Scldel_Sdadel(uint32_t clock_src_freq, uint32_t I2C_speed);
uint32_t i2c_compute_Scll_Sclh(uint32_t clock_src_freq, uint32_t I2C_speed);
```

Reply:
`i2c_device.h` belongs to HAL implementation? similar to i2c_api.c so it should be consistent with the naming.  Mbed OS style guide specifies to use lower case with `_` as a separator. 

If this is the case (this is HAL implementation), ti would be:

```
uint32_t i2c_compute_timing(uint32_t clock_src_freq, uint32_t i2c_freq);
uint32_t i2c_get_timing(I2CName i2c, int hz);
void i2c_compute_presc_scldel_sdadel(uint32_t clock_src_freq, uint32_t i2c_speed);
uint32_t i2c_compute_scll_sclh(uint32_t clock_src_freq, uint32_t i2c_speed);
```